### DISCUSSION TITLE
All I wanted was a banana ...

### DISCUSSION THREAD

**Discussion by pkese** [astral-sh/rye#484] [General]
_... but what I got was a gorilla holding the banana and the entire jungle._

Rye claims itself to be "An Experimental **Package Management** Solution for Python" so I wanted to create an reproducible Python environment where I could run a Jupyter Notebook.

I typed `rye init myproject` and instead of a lightweight Python virtual environment where I could install packages, Rye gave me a whole project structure with opinionated subdirectories, Git, README and a bunch of lock files.

Please stop advertising Rye as a package manager (banana) if you're actually selling a project tool (gorilla with a jungle).

Also ... If lock files are supposed to be machine generated rather than edited manually, I really don't see a compelling reason why you need three of them. A single one should suffice, anything else is just polluting root source directory with unnecessary files.

**Comment by mitsuhiko** [comment]
I don't think that Rye is advertised in any particular way. I think I'm quite clear on it being an experimental project that codifies my personal needs:

> Rye is [Armin's](https://github.com/mitsuhiko/) personal one-stop-shop for all his Python needs. It installs and manages Python installations, manages pyproject.toml files, installs and uninstalls dependencies, manages virtualenvs behind the scenes. It supports monorepos and global tool installations.
